library/src/main/java/com/google/android/exoplayer/MediaCodecUtil.java

```diff
@@ -95,34 +105,72 @@ public static synchronized void warmCodec(String mimeType, boolean secure) {
   /**
    * Returns the name of the best decoder and its capabilities for the given mimeType.
    *
-   * @param mimeType The mime type.
+   * @param originalMimeType The mime type.
    * @param secure Whether the decoder is required to support secure decryption. Always pass false
    *     unless secure decryption really is required.
    * @return The name of the best decoder and its capabilities for the given mimeType, or null if
    *     no decoder exists.
    */
   public static synchronized Pair<String, CodecCapabilities> getMediaCodecInfo(
-      String mimeType, boolean secure) throws DecoderQueryException {
-    CodecKey key = new CodecKey(mimeType, secure);
-    if (codecs.containsKey(key)) {
-      return codecs.get(key);
-    }
-    MediaCodecListCompat mediaCodecList = Util.SDK_INT >= 21
-        ? new MediaCodecListCompatV21(secure) : new MediaCodecListCompatV16();
-    Pair<String, CodecCapabilities> codecInfo = getMediaCodecInfo(key, mediaCodecList);
-    // TODO: Verify this cannot occur on v22, and change >= to == [Internal: b/18678462].
-    if (secure && codecInfo == null && Util.SDK_INT >= 21) {
-      // Some devices don't list secure decoders on API level 21. Try the legacy path.
-      mediaCodecList = new MediaCodecListCompatV16();
+      String originalMimeType, boolean secure) throws DecoderQueryException {
```

Comment:
This is quite a bit more invasive than I had envisioned. Are there any devices other than Minix Neo that are using the wrong mimeTypes, or is this *just* for that device?